server/api/users.js

```diff
@@ -1,14 +1,40 @@
 const router = require('express').Router()
-const {User} = require('../db/models')
+const { User, Teachable } = require('../db/models');
 module.exports = router
 
 router.get('/', (req, res, next) => {
-  User.findAll({
-    // explicitly select only the id and email fields - even though
-    // users' passwords are encrypted, it won't help if we just
-    // send everything to anyone who asks!
-    attributes: ['id', 'email']
-  })
-    .then(users => res.json(users))
+  User.findAll({ include: [{ model: Teachable }] })
+    .then(users => res.status(200).json(users))
     .catch(next)
 })
+
+router.get('/:id', (req, res, next) => {
+  User.findById(req.params.id)
+    .then(user => res.status(200).json(user))
+    .catch(next);
+})
+
+//Get All users by a given Teachable
+router.get('/:teachable/', (req, res, next) => {
+  const teachable = req.params.teachable; //Assuming this is an id number? Also works if stored as string?
+  User.findAll({ include: [{ model: Teachable }] })
```

Comment:
Hmm. I have uneasy Feelings about this query-then-filter strategy but will suppress them for now. I may ask you to explain your thinking to me tomorrow.

Reply:
We weren't sure if we could use a where here, since Teachable has to be included in, and teachables is an array we have to search through to find one Teachable. We could use advice, since we're not sure if our approach even works. Maybe we should be using the join table for this instead?